app/pages/home-for-user/recent-projects.jsx

```diff
@@ -58,14 +63,17 @@ const RecentProjectsSection = React.createClass({
       });
 
       return Promise.all(projects.map((project) => {
-        return project.get('avatar')
-        .catch(() => {
-          return null;
-        })
-        .then((avatar) => {
-          this.state.avatars[project.id] = avatar;
-          this.forceUpdate();
-        });
+        if (project.links.avatar.id) {
+          return apiClient.type('avatars')
+          .get(project.links.avatar.id)
+          .catch(() => {
+            return null;
+          })
+          .then((avatar) => {
+            this.state.avatars[project.id] = avatar;
```

Comment:
I have no idea what's going on here. @srallen is right, state should be changed via `this.setState`.